### PR TITLE
Fix some typos

### PR DESCRIPTION
**Pull request by gkoz**
Hi!

I liked reading this and felt like contributing some proofreading if you don't mind.

Of the things I didn't touch, why "an `&mut`"?

### CODE REVIEW

**Comment by Gankro** [comment]
Yeah I'm weird I read & as "and".

File: text/first-pop.md [hunk 1]
```diff
@@ -197,12 +197,12 @@ We should probably step back and think about what we're trying to do. We want
...
-    * remove the head of the list
-    * remove its `elem`
-    * replace the lists head with its `next`
-    * return `Some(elem)`
+* If it's empty, just return None.
+* It it's *not* empty,
+    * remove the head of the list,
+    * remove its `elem`,
+    * replace the lists head with its `next`,
+    * return `Some(elem)`.
```
Comment (Gankro) [inline text/first-pop.md]: I'd rather not have the punctuation here if that's ok with you. I usually only punctuate longer-winded lists.

File: text/first-push.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ impl List {
 }
 ```
 
-First thing's first, we need to make a node to store our element in:
+First things first, we need to make a node to store our element in:
```
Comment (Gankro) [inline text/first-push.md]: Hmm... I thought this was correct "first thing's first" -> "first thing is first"

Comment (gkoz) [inline text/first-push.md]: [This](http://english.stackexchange.com/questions/7419/which-is-the-correct-idiom-first-things-first-or-first-things-first) gives a good overview.

**Comment by gkoz** [comment]
Removed the list punctuation diff.

**Comment by Gankro** [comment]
Thanks!

**Pull request merged**